fix: resolve Windows MCP server execution for script-based tools

25621 opened the pull request:
## What?

Fixes MCP server initialization failures on Windows when using script-based tools like `npx`, `pnpm`, and `yarn` that rely on `.cmd`/`.bat` files rather than `.exe` binaries.

Fixes #2945

## Why?

Windows users encounter "program not found" errors when configuring MCP servers with commands like `npx` in their `~/.codex/config.toml`. This happens because:

- Tools like `npx` are batch scripts (`npx.cmd`) on Windows, not executable binaries
- Rust's `std::process::Command` bypasses the shell and cannot execute these scripts directly
- The Windows shell normally handles this by checking `PATHEXT` for executable extensions

Without this fix, Windows users must specify full paths or add `.cmd` extensions manually, which breaks cross-platform compatibility.

## How?

Added platform-specific program resolution using the `which` crate to find the correct executable path:

- **Windows**: Resolves programs through PATH/PATHEXT to find `.cmd`/`.bat` scripts
- **Unix**: Returns the program unchanged (no-op, as Unix handles scripts natively)

### Changes

- Added `which = "6"` dependency to `mcp-client/Cargo.toml`
- Implemented `program_resolver` module in `mcp_client.rs` with platform-specific resolution
- Added comprehensive tests for both Windows and Unix behavior

### Testing

Added platform-specific tests to verify:
- Unix systems execute scripts without extensions
- Windows fails without proper extensions
- Windows succeeds with explicit extensions
- Cross-platform resolution enables successful execution

**Tested on:**
- Windows 11 (NT 10.0.26100.0 x64)
- PowerShell 5.1 & 7+, CMD, Git Bash
- MCP servers: playwright, context7, supabase
- WSL (verified no regression)

**Local checks passed:**
```bash
cargo test && cargo clippy --tests && cargo fmt -- --config imports_granularity=Item
```

### Results

**Before:**
```
🖐 MCP client for `playwright` failed to start: program not found
```

**After:**
```
🖐 MCP client for `playwright` failed to start: request timed out
```

Windows users can now use simple commands like `npx` in their config without specifying full paths or extensions. The timeout issue is a separate concern that will be addressed in a follow-up PR.

25621 committed:
fix(rmcp-client): resolve Windows MCP server execution for script-based tools

This ports the fix from the original PR #3828 to rmcp-client,
which replaced mcp-client in PR #5529.

On Windows, script-based MCP servers (e.g., npx, pnpm, yarn) that
rely on .cmd or .bat files fail to execute. This happens because
Rust's Command does not resolve scripts via PATHEXT like the
shell does.

This fix uses the which crate to resolve the full executable path
(including .cmd/.bat extensions) before spawning the process,
allowing these tools to run correctly.

Changes:
- Add which crate dependency to rmcp-client/Cargo.toml
- Implement program_resolver module with platform-specific logic:
  * Windows: Resolves .cmd/.bat scripts using which::which_in
  * Unix: No-op (OS handles shebangs natively)
- Modify new_stdio_client to resolve program path before Command::new
- Add unit tests for program resolution on both platforms

Fixes: #2945

25621 committed:
Merge branch 'main' into fix/windows-mcp-script-resolution

dylan-hurd-oai commented on codex-rs/rmcp-client/src/rmcp_client.rs:
```diff
@@ -412,3 +419,214 @@ async fn create_oauth_transport_and_runtime(
+    #[cfg(unix)]
+    pub fn resolve(program: OsString, _env: &HashMap<String, String>) -> std::io::Result<OsString> {
+        Ok(program)
+    }
```
nit: can we move this one to be first? It's simpler / shorter.

dylan-hurd-oai reviewed (commented):
Thanks for the contribution! 2 small organizational comments to address, but overall this looks good to me

25621 committed:
refactor(rmcp-client): move program_resolver to separate module file (#3828)

Per review feedback, extracting program_resolver module to its own file
for better code organization and maintainability.

- Move program_resolver mod to src/program_resolver.rs
- Reorder Unix/Windows functions (Unix first as simpler)
- Move related tests to the new module file
- Add module declaration to lib.rs

etraut-openai committed:
Merge branch 'main' into fix/windows-mcp-script-resolution
